hive/runners/multi_agent_loop.py

```diff
@@ -124,159 +178,40 @@ def run_end_step(self, episode_metrics, terminated=False, truncated=False):
 
         """
         for agent in self._agents:
-            if self._transition_info.is_started(agent):
-                info = self._transition_info.get_info(agent, terminated, truncated)
+            if transition_info.is_started(agent):
+                info = transition_info.get_info(agent, terminated, truncated)
+
                 if self._training:
                     agent.update(info)
                 episode_metrics[agent.id]["episode_length"] += 1
                 episode_metrics["full_episode_length"] += 1
             episode_metrics[agent.id]["reward"] += info["reward"]
 
-    def run_episode(self):
+    def run_episode(self, environment):
         """Run a single episode of the environment."""
         episode_metrics = self.create_episode_metrics()
-        observation, turn = self._environment.reset()
-        self._transition_info.reset()
+        observation, turn = environment.reset()
+        transition_info = TransitionInfo(self._agents, self._stack_size)
         steps = 0
         terminated, truncated = False, False
 
         # Run the loop until the episode ends or times out
-        while not (terminated or truncated) and steps < self._max_steps_per_episode:
+        while (
+            not (terminated or truncated)
+            and steps < self._max_steps_per_episode
+            and (not self._training or self._train_schedule.get_value())
+        ):
             terminated, truncated, observation, turn = self.run_one_step(
```

Comment:
This comment is related to gym new update PR. I guess we are assuming that all the environments including single/multi-agent return **truncated**, right? I believe the way it's implemented now is going to break Petting Zoo envs at least since env.last() which is called in env.step() does not return truncated. Just raising this here so we add an issue maybe to fix those envs.

Reply:
Since we don't have test_ma_loop yet, I just suspect that we might miss something. Could you just run a simple experiment to verify that also runs with no bug? Thanks.

Reply:
The new pettingzoo API does return truncated: https://pettingzoo.farama.org/content/basic_usage/. And yes I think we already ran basic multiagent experiments to make sure everything runs.